In newer systems (Debian 11) gettid() is provided by unistd if using GNU extensions.
This patch checks for __GNU extension and removes gettid() inline from compilation.

--- src/lib/atomic.h.orig	2022-04-23 22:42:46.350826799 +0200
+++ ./src/lib/atomic.h	2022-04-23 22:50:50.271144861 +0200
@@ -89,6 +89,7 @@
 
 
 /* returns (linux-specific) thread-id (for single-thread processes it's just PID) */
+#ifndef __USE_GNU
 static inline uint32_t gettid() {
   /* AUTO_TSC("gettid()"); */
 #ifdef __linux__
@@ -107,6 +108,7 @@
   return tid;
 #endif
 }
+#endif
 
 #ifdef __cplusplus
 }
